test-dts/defineComponent.test-d.tsx

```diff
@@ -129,7 +130,7 @@ describe('with object props', () => {
       expectType<ExpectedProps['eee']>(this.eee)
       expectType<ExpectedProps['fff']>(this.fff)
 
-      // props on `this` should be readonly
+      // @ts-expect-error props on `this` should be readonly
       expectError((this.a = 1))
```

Comment:
is `expectError` necessery here? maybe it's enough to do
```ts
// @ts-expect-error props on `this` should be readonly
this.a = 1
```
?

Reply:
I kept the `expectError` to keep the same interface as before, felt a bit better to me, is not necessary to have it